Content/NPCs/Misc/MagnetizedEnemies.cs

```diff
@@ -0,0 +1,278 @@
+                    {
+                        break;
+                    }
+                }
+
+                if (attackCounter % attackCycleLength > 250 && attackCounter % attackCycleLength < 260)
+                    flashOpacityPreEase += 0.1f;
+                else if (attackCounter % attackCycleLength > 260 && attackCounter % attackCycleLength < 270)
+                    flashOpacityPreEase -= 0.1f;
+
+                flashOpacityPreEase = MathHelper.Clamp(flashOpacityPreEase, 0, 1);
+
+                flashOpacity = EaseFunction.EaseCircularIn.Ease(flashOpacityPreEase);
+
+
+                if (attackCounter % attackCycleLength == 300)
+                {
+                    Helper.PlayPitched("Magic/LightningExplode", 0.4f, 0f, npc.Center);
+                    if (cache != null)
+                    {
+                        for (int i = 2; i < cache.Count - 2; i++)
+                        {
+                            Vector2 vel = cache[i].DirectionTo(npc.Center).RotatedByRandom(0.4f) * Main.rand.NextFloat(2, 4);
+                            Dust.NewDustPerfect(cache[i] + Main.rand.NextVector2Circular(16, 16), ModContent.DustType<GlowLine>(), vel, 0, Color.Cyan, 0.8f);
+                        }
+                    }
+                    Projectile.NewProjectile(npc.GetSource_FromAI(), npc.Center, Vector2.Zero, ModContent.ProjectileType<EnemyMagnetShot>(), (int)(npc.damage * (Main.expertMode || Main.masterMode ? 0.5f : 1)), 0, default, endPoint.X, endPoint.Y);
+                    fade = 4.5f;
+                }
+
+                endPoint = npc.Center + offset;
+
+                if (!Main.dedServ)
+                {
+                    ManageCache(npc);
+                    ManageTrails(npc);
+                }
+
+                if (fade > 1f)
+                {
+                    fade -= 0.1f;
+                    cache?.ForEach(n => Lighting.AddLight(n, Color.Cyan.ToVector3() * ((fade - 1) * 0.3f)));
+                }
+            }
+        }
+
+        public override void OnKill(NPC npc)
+        {
+            if (chargedPlayer == default)
+                return;
+            for (int index = 0; index < 54; ++index)
+            {
+                if (chargedPlayer.inventory[index].type == ModContent.ItemType<UnchargedMagnet>() && chargedPlayer.inventory[index].stack > 0)
+                {
+                    chargedPlayer.inventory[index].stack--;
+                    if (chargedPlayer.inventory[index].stack <= 0)
+                        chargedPlayer.inventory[index].TurnToAir();
+
+                    Item.NewItem(npc.GetSource_Death(), chargedPlayer.Center, ModContent.ItemType<ChargedMagnet>());
+
+                    break;
+                }
+            }
+        }
+
+        public override bool PreDraw(NPC npc, SpriteBatch spriteBatch, Vector2 screenPos, Color drawColor)
+        {
+            if (charged)
+                DrawPrimitives();
```

Comment:
linebreak after this would probably be nicer on the eyes

Reply:
fixed